### PR TITLE
Spark 3.5: Fix comment and assertion mismatch in PartitionedWritesTestBase/TestRewritePositionDeleteFilesAction

### LINKED ISSUE #11747: Comment and assertion mismatch in PartitionedWritesTestBase/TestRewritePositionDeleteFilesAction
### Apache Iceberg version

main (development)

### Query engine

Spark

### Please describe the bug 🐞

There is a mismatch between the test comment and the assertion in the following unit test, which might lead to confusion or logical errors.

Here is the relevant code snippet:

```java
  @TestTemplate
  public void testInsertAppend() {
    assertThat(scalarSql("SELECT count(*) FROM %s", selectTarget()))
        .as("Should have 5 rows after insert")
        .isEqualTo(3L);
```
- The comment "Should have 5 rows after insert" indicates the expectation of 5 rows in the table.
- However, the actual assertion isEqualTo(3L) checks for 3 rows instead.

### Willingness to contribute

- [X] I can contribute a fix for this bug independently
- [X] I would be willing to contribute a fix for this bug with guidance from the Iceberg community
- [ ] I cannot contribute a fix for this bug at this time

### PR DESCRIPTION
**Pull request by wzx140**
close #11747

### CODE REVIEW

**Comment by ebyhr** [comment]
Can we fix `TestRewritePositionDeleteFilesAction` as well? 

https://github.com/apache/iceberg/blob/ff813445916bfd6ec1cc30a02b02f8bade7a26f6/spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewritePositionDeleteFilesAction.java#L278

https://github.com/apache/iceberg/blob/ff813445916bfd6ec1cc30a02b02f8bade7a26f6/spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewritePositionDeleteFilesAction.java#L472

**Comment by wzx140** [comment]
> Can we fix `TestRewritePositionDeleteFilesAction` as well?
> 
> https://github.com/apache/iceberg/blob/ff813445916bfd6ec1cc30a02b02f8bade7a26f6/spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewritePositionDeleteFilesAction.java#L278
> 
> https://github.com/apache/iceberg/blob/ff813445916bfd6ec1cc30a02b02f8bade7a26f6/spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewritePositionDeleteFilesAction.java#L472

@ebyhr done

**Review by huaxingao — APPROVED** [review]

**Comment by huaxingao** [comment]
cc @szehon-ho

File: spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewritePositionDeleteFilesAction.java [hunk 1]
```diff
@@ -275,7 +275,7 @@ public void testRewriteFilter() throws Exception {
             .execute();
 
     List<DeleteFile> newDeleteFiles = except(deleteFiles(table), deleteFiles);
-    assertThat(newDeleteFiles).as("Should have 4 delete files").hasSize(2);
+    assertThat(newDeleteFiles).as("Should have 2 delete files").hasSize(2);
```
Comment (szehon-ho) [inline spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewritePositionDeleteFilesAction.java:278]: Question, should we just remove the number here, how about, 'delete files'?

Wont the assert message in the end already have the expected value from hasSize()?

Comment (wzx140) [reply spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewritePositionDeleteFilesAction.java:278]: @ebyhr @huaxingao @szehon-ho  I think this suggestion is reasonable, but there are many similar assertions in the existing uts. Should we work together to update them as well?

Comment (wzx140) [reply spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewritePositionDeleteFilesAction.java]: Done. The updated assert message is as follows:

[Delete files] 
Expected size: 1 but was: 2 in:

**Review by szehon-ho — APPROVED** [review]

File: spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/sql/PartitionedWritesTestBase.java [hunk 2]
```diff
@@ -53,7 +53,7 @@ public void removeTables() {
   @TestTemplate
   public void testInsertAppend() {
     assertThat(scalarSql("SELECT count(*) FROM %s", selectTarget()))
-        .as("Should have 5 rows after insert")
+        .as("Rows after insert")
```
Comment (ebyhr) [inline spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/sql/PartitionedWritesTestBase.java:56]: The original message seems wrong. I think it should be "Rows **_before_** insert". 
Same for `testInsertOverwrite`.

Comment (szehon-ho) [reply spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/sql/PartitionedWritesTestBase.java:56]: Good catch

Comment (wzx140) [reply spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/sql/PartitionedWritesTestBase.java]: done

**Comment by szehon-ho** [comment]
Merged, thanks @wzx140 and @ebyhr @huaxingao for detailed review, feel free to open follow up prs to fix the other spark versions, other tests, etc.

**Pull request merged**